video/out/vulkan/context_display.c

```diff
@@ -0,0 +1,423 @@
+/*
+ * This file is part of mpv.
+ *
+ * mpv is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * mpv is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "options/m_config.h"
+
+#include "context.h"
+#include "utils.h"
+
+struct vulkan_display_opts {
+    char *display_spec;
+};
+
+static bool parse_display_spec(struct mp_log *log,
+                               const char *display_spec,
+                               int *display, int *mode, int *plane)
+{
+    int parsed_display = 0;
+    int parsed_mode = 0;
+    int parsed_plane = 0;
+    bool ret = false;
+
+    if (!display_spec) {
+        goto finish;
+    }
+    const char *mode_ptr = strchr(display_spec, ':');
+    if (mode_ptr) {
+        parsed_display = atoi(display_spec);
+        const char *plane_ptr = strchr(mode_ptr + 1, ':');
+        if (plane_ptr) {
+            parsed_mode = atoi(mode_ptr + 1);
+        }
+        if (*(plane_ptr + 1) != '\0') {
```

Comment:
Segfault if `plane_ptr` is NULL. As written, this needs to be inside the previous `if`. (Also, I think `plane_ptr[1]` is more readable than `*(plane_ptr + 1)`)